src/NLog/Targets/EntityFrameworkTarget.cs

```diff
@@ -0,0 +1,134 @@
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__
+
+namespace NLog.Targets
+{
+
+    using System.Data.Common;
+    using NLog.Common;
+    using NLog.Layouts;
+
+    /// <summary>
+    /// An NLog target that writes logging messages using an ADO.NET provider that is configurable via an Entity Framework connection string.
+    /// </summary>
+    [Target("EntityFramework")]
+    public class EntityFrameworkTarget : DatabaseTarget
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EntityFrameworkTarget" /> class.
+        /// </summary>
+        public EntityFrameworkTarget()
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EntityFrameworkTarget" /> class.
+        /// </summary>
+        /// <param name="name">Name of the target.</param>
+        public EntityFrameworkTarget(string name) : this() => Name = name;
+
+        /// <summary>
+        /// Initializes the target. Can be used by inheriting classes
+        /// to initialize logging.
+        /// </summary>
+        protected override void InitializeTarget()
+        {
+#if !NETSTANDARD
+            base.InitializeTarget();
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (UseTransactions.HasValue)
+#pragma warning restore CS0618 // Type or member is obsolete
+            {
+                InternalLogger.Warn("UseTransactions is obsolete and will not be used - will be removed in NLog 6");
+            }
+
+            if (string.IsNullOrEmpty(ConnectionStringName))
+            {
+                throw new NLogConfigurationException("ConnectionStringName is required parameter.");
+            }
+            // read connection string and provider factory from entity framework connection string
+            var cs = this.ConnectionStringsSettings[ConnectionStringName];
+            if (cs == null)
+            {
+                throw new NLogConfigurationException("Connection string '" + ConnectionStringName + "' is not declared in <connectionStrings /> section.");
+            }
+
+            const string providerConnectionStringParameterName = "provider connection string";
+            const string providerParameterName = "provider";
+
+            var dbConnectionStringBuilder = new DbConnectionStringBuilder { ConnectionString = cs.ConnectionString };
+
+            //CurrentValues Dictionary in DbConnectionStringBuilder is case insensitive (StringComparer.OrdinalIgnoreCase)
+
+            //should we use SimpleLayout.Escape here?
+            string provider = (string)dbConnectionStringBuilder[providerParameterName];
```

Comment:
`SimpleLayout.Escape` isn't needed here as provider is a `string` (and not a Layout)